server/controllers/finance/reports/cashflow/index.js

```diff
@@ -66,11 +69,12 @@ function processingCashflowReport(params) {
         throw new BadRequest('Periods not found due to a bad date interval', 'ERRORS.BAD_DATE_INTERVAL');
       }
       glb.periods = periods;
-      return closingBalance(params.account_id, glb.periods[0].start_date);
+      return AccountsExtra.getOpeningBalanceForDate(params.account_id, glb.periods[0].start_date);
     })
     .then((balance) => {
-      if (!balance.length) { balance[0] = { balance : 0, account_id : params.account_id }; }
-      glb.openningBalance = balance[0];
+      if (!balance) { balance = { balance : 0, account_id : params.account_id }; }
```

Comment:
`AccountsExtra.getOpeningBalanceForDate()` will return an object.  See [this line](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/server/controllers/finance/accounts/extra.js#L152).  That line does not contain a `account_id`.

Is there ever a case where `if(!balance) {}` is triggered?  Do you ever use the `account_id`?